### PR TITLE
fix sync issue with theme

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

Docusaurus is buggy when it comes to syncing the user's preferred theme. Made a custom hook that should hopefully do better.

We still need the `respectPrefersColorScheme` since that is what is syncing the theme-toggle-button's state it looks like.

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/src/pages/index.tsx [hunk 1]
```diff
@@ -36,11 +36,34 @@ function useLocalStorageVersion() {
   };
 }
 
+const usePrefferedTheme = () => {
```
Comment (KATT) [inline www/src/pages/index.tsx:39]: Preferred

File: www/src/pages/index.tsx [hunk 2]
```diff
@@ -36,11 +36,34 @@ function useLocalStorageVersion() {
...
+  const lightTheme = 'light';
+
+  React.useEffect(() => {
+    const mediaMatch = window.matchMedia('(prefers-color-scheme: dark)');
+    const htmlElement = document.querySelector('html');
+
+    const colorSchemeChangeListener = (e: MediaQueryListEvent) => {
+      const newTheme = e.matches ? darkTheme : lightTheme;
+      htmlElement?.setAttribute('data-theme', newTheme);
+      console.log('setting theme to', newTheme);
```
Comment (KATT) [inline www/src/pages/index.tsx:50]: Rm log

**Review by KATT — APPROVED** [review]
Maybe you can get this into docusaurus core?

**Comment by juliusmarminge** [comment]
This seems reliable on the homepage. Will try to find a way to make it sync on the docs pages too

**Review by bautistaaa — APPROVED** [review]
docusaurus is wonky

**Comment by juliusmarminge** [comment]
> docusaurus is wonky

Yea I'll try on a fresh docu and see if I can get something into core perhaps 🤔

**Review by KATT — APPROVED** [review]

**Pull request merged**